src/Neo/SmartContract/Native/ContractEventAttribute.cs

```diff
@@ -17,35 +17,48 @@ namespace Neo.SmartContract.Native
 {
     [DebuggerDisplay("{Descriptor.Name}")]
     [AttributeUsage(AttributeTargets.Constructor, AllowMultiple = true)]
-    internal class ContractEventAttribute : Attribute
+    internal class ContractEventAttribute : Attribute, IHardforkActivable
     {
         public int Order { get; init; }
         public ContractEventDescriptor Descriptor { get; set; }
         public Hardfork? ActiveIn { get; init; } = null;
+        public Hardfork? DeprecatedIn { get; init; } = null;
```

Comment:
You may apply https://github.com/neo-project/neo/pull/3234#discussion_r1602806943 if you'd like to, as a style unification.

Reply:
I think that should be applied in another pr.